ecs-agent/tcs/handler/handler_test.go

```diff
@@ -340,32 +352,103 @@ func TestConnectionInactiveTimeout(t *testing.T) {
 		testAgentVersion,
 		testAgentHash,
 		testContainerRuntimeVersion,
-		server.URL,
 		false,
 		testCreds,
 		testCfg,
 		deregisterInstanceEventStream,
-		50*time.Millisecond,
+		5*time.Second,
 		100*time.Millisecond,
 		testDisconnectionTimeout,
 		testDisconnectionJitter,
 		nil,
 		telemetryMessages,
 		healthMessages,
 		emptyDoctor,
+		testecsclient,
 	)
 
 	// Start a session with the test server.
-	err = session.StartTelemetrySession(ctx, server.URL)
+	err = session.StartTelemetrySession(ctx)
 
-	assert.NoError(t, err, "Close the connection should cause the tcs client return error")
+	// TODO: confirm once. The client would not attempt to reconnect since we are using StartTelemetrySession
+	// instead of Start. Modified the test to just test for connection closed on inactivity.
+	//assert.NoError(t, err, "Close the connection should cause the tcs client return error")
```

Comment:
Should remove this todo? I think what it says about is done in `TestClientReconnectsAfterInactiveTimeout`?

Reply:
oh yeah sure. Just wanted to confirm that if I am understanding it correctly. Looks like it is correct and also I have added another testcase to test the reconnects.